### PR TITLE
Fix for lockscreen shown after user dismissal bugs when displayMode is always

### LINKED ISSUE #1826: Lockscreen shown after user dismissal
## Bug Report
There are four bugs with the lockscreen being shown after being dismissed by the user if a policy table update changes `lockScreenDismissalEnabled` in the `OnDriverDistraction` from `true` to `false`. 

### Reproduction Steps for Bug 1
1. LockScreenConfiguration.displayMode has been set to `always`.
2. The last received `OnDriverDistraction.lockScreenDismissalEnabled` from the module was `true`. The user swiped to dismiss the lockscreen.
3. The module sends a `OnDriverDistraction` notification with `state = OFF` and `lockScreenDismissalEnabled` to `null`.

##### Expected Behavior
Lockscreen is not shown.

##### Observed Behavior
Locks screen is shown with the `OnDriverDistraction.lockScreenDismissalWarning` message. User can swipe to dismiss the lockscreen.

### Reproduction Steps for Bug 2
1. LockScreenConfiguration.displayMode has been set to `always`.
2. The last received `OnDriverDistraction.lockScreenDismissalEnabled` from the module was `true`. The user swiped to dismiss the lockscreen.
3. The module sends a `OnDriverDistraction` notification with `state = OFF` and `lockScreenDismissalEnabled` to `false`.

##### Expected Behavior
Lockscreen is shown with the `OnDriverDistraction.lockScreenDismissalWarning` message. User **cannot** swipe to dismiss the lockscreen.

##### Observed Behavior
Locks screen is shown with the `OnDriverDistraction.lockScreenDismissalWarning` message. User can swipe to dismiss the lockscreen.

### Reproduction Steps for Bug 3
1. LockScreenConfiguration.displayMode has been set to `always`.
2. The last received `OnDriverDistraction.lockScreenDismissalEnabled` from the module was `true`. The user swiped to dismiss the lockscreen.
3. The module sends a `OnDriverDistraction` notification with `state = ON` and `lockScreenDismissalEnabled` to `null`.

##### Expected Behavior
Lockscreen is not shown.

##### Observed Behavior
Locks screen is shown with the `OnDriverDistraction.lockScreenDismissalWarning` message. User can swipe to dismiss the lockscreen.

### Reproduction Steps for Bug 4
1. LockScreenConfiguration.displayMode has been set to `always`.
2. The last received `OnDriverDistraction.lockScreenDismissalEnabled` from the module was `true`. The user swiped to dismiss the lockscreen.
3. The module sends a `OnDriverDistraction` notification with `state = ON` and `lockScreenDismissalEnabled` to `true`.

##### Expected Behavior
Lockscreen is not shown.

##### Observed Behavior
Locks screen is shown with the `OnDriverDistraction.lockScreenDismissalWarning` message. User can swipe to dismiss the lockscreen.

### OS & Version Information
* iOS Version: n/a
* SDL iOS Version: 7.0
* Testing Against: sdl_hmi and sdl_core.

### PR DESCRIPTION
**Pull request by FrankElias77**
Fixes #1826 

### Risk
This PR makes **no** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [ ] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Unit Tests
N/A

#### Core Tests
N/A

Core version / branch / commit hash / module tested against: N/A
HMI name / version / branch / commit hash / module tested against: N/A

### Summary
Fix for specific cases when the lockScreen's displayMode is set to .always. The lockScreen was being presented after it has been previously dismissed by user.

##### Breaking Changes
* N/A

##### Enhancements
* N/A

##### Bug Fixes
* Fix for specific cases when the lockScreen's displayMode is set to .always. The lockScreen was being presented after it has been previously dismissed by user.

### Tasks Remaining:
N/A

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/private/SDLLockScreenManager.m [hunk 1]
```diff
@@ -212,11 +211,13 @@ - (void)sdl_updateLockScreenDismissable {
...
-        [self.lastDriverDistractionNotification.state isEqualToEnum:SDLDriverDistractionStateOn] &&
-        !self.lockScreenDismissable) {
-        self.lockScreenDismissedByUser = NO;
+
+    if (!(self.config.displayMode == SDLLockScreenConfigurationDisplayModeAlways)) {
+        if (self.lockScreenDismissedByUser &&
+            [self.lastDriverDistractionNotification.state isEqualToEnum:SDLDriverDistractionStateOn] &&
+            !self.lockScreenDismissable) {
+            self.lockScreenDismissedByUser = NO;
+        }
```
Comment (joeljfischer) [inline SmartDeviceLink/private/SDLLockScreenManager.m:220]: I think this could just be another && addition

**Review by joeljfischer — CHANGES_REQUESTED** [review]

**Review by NicoleYarroch — CHANGES_REQUESTED** [review]
The bugs reported in the issue have been fixed. However, a new bug was introduced. The details for reproducing are below: 

1. LockScreenConfiguration.displayMode has been set to `always`.
2. The last received `OnDriverDistraction.lockScreenDismissalEnabled` from the module was `true`. The user swiped to dismiss the lockscreen.
3. The module sends a `OnDriverDistraction` notification with `state = ON` and `lockScreenDismissalEnabled` to `false`.

##### Expected Behavior
Lock screen is shown (without the `OnDriverDistraction.lockScreenDismissalWarning`) and the user can not swipe to dismiss the lock screen.

##### Observed Behavior
Locks screen is not shown.

**Review by NicoleYarroch — CHANGES_REQUESTED** [review]
It looks like bug #3 in the issue report is now broken after the latest changes were made.

**Review by NicoleYarroch — CHANGES_REQUESTED** [review]
Smoke testing went well but I made a request for more tests to be added so we don't accidentally break the code in the future.

File: SmartDeviceLink/private/SDLLockScreenManager.m [hunk 2]
```diff
@@ -212,11 +214,15 @@ - (void)sdl_updateLockScreenDismissable {
...
-    if (self.lockScreenDismissedByUser &&
-        [self.lastDriverDistractionNotification.state isEqualToEnum:SDLDriverDistractionStateOn] &&
-        !self.lockScreenDismissable) {
-        self.lockScreenDismissedByUser = NO;
+
+    if (!(self.config.displayMode == SDLLockScreenConfigurationDisplayModeAlways)) {
+        if (self.lockScreenDismissedByUser &&
+            [self.lastDriverDistractionNotification.state isEqualToEnum:SDLDriverDistractionStateOn] &&
+            !self.lockScreenDismissable) {
+            self.lockScreenDismissedByUser = NO;
```
Comment (NicoleYarroch) [inline SmartDeviceLink/private/SDLLockScreenManager.m:222]: Can you also add tests to **SDLLockScreenManagerSpec** test this line?

File: SmartDeviceLink/private/SDLLockScreenManager.m [hunk 3]
```diff
@@ -212,11 +214,15 @@ - (void)sdl_updateLockScreenDismissable {
...
+        }
+    } else if (self.config.displayMode == SDLLockScreenConfigurationDisplayModeAlways) {
+        [self sdl_updateLockscreenViewControllerWithDismissableState:self.lockScreenDismissable];
```
Comment (NicoleYarroch) [inline SmartDeviceLink/private/SDLLockScreenManager.m:225]: Can you also add tests to **SDLLockScreenManagerSpec** test this line?

**Review by NicoleYarroch — CHANGES_REQUESTED** [review]
Left some comments

File: SmartDeviceLinkTests/DevAPISpecs/SDLLockScreenManagerSpec.m [hunk 4]
```diff
@@ -432,6 +434,8 @@ @interface SDLLockScreenManager ()
                 OCMVerifyAllWithDelay(fakeViewControllerPresenter, 0.5);
 
                 expect(((SDLFakeViewControllerPresenter *)fakeViewControllerPresenter).shouldShowLockScreen).toEventually(beTrue());
+
+                expect(testManager.lockScreenDismissable).to(equal(NO));
```
Comment (NicoleYarroch) [inline SmartDeviceLinkTests/DevAPISpecs/SDLLockScreenManagerSpec.m:438]:
```suggestion
                expect(testManager.lockScreenDismissable).to(beFalse());
```

**Review by NicoleYarroch — CHANGES_REQUESTED** [review]
Left some comments

**Comment by joeljfischer** [comment]
These issues are fixed on develop

**Pull request closed without merging**